src/build/functions/edge.ts

```diff
@@ -85,13 +85,27 @@ const writeHandlerFile = async (ctx: PluginContext, { matchers, name }: NextDefi
     JSON.stringify(minimalNextConfig),
   )
 
+  const htmlRewriterWasm = await readFile(
+    join(
+      ctx.pluginDir,
+      'edge-runtime/vendor/deno.land/x/htmlrewriter@v1.0.0/pkg/htmlrewriter_bg.wasm',
+    ),
+  )
+
   // Writing the function entry file. It wraps the middleware code with the
   // compatibility layer mentioned above.
   await writeFile(
     join(handlerDirectory, `${handlerName}.js`),
     `
+    import { decode as _base64Decode } from './edge-runtime/vendor/deno.land/std@0.175.0/encoding/base64.ts';
+    import { init as htmlRewriterInit } from './edge-runtime/vendor/deno.land/x/htmlrewriter@v1.0.0/src/index.ts'
     import {handleMiddleware} from './edge-runtime/middleware.ts';
     import handler from './server/${name}.js';
+
+    await htmlRewriterInit({ module_or_path: _base64Decode(${JSON.stringify(
+      htmlRewriterWasm.toString('base64'),
+    )}).buffer });
+
```

Comment:
bundling wasm file as-is doesn't seem to work, so instead we inline it using same method we do for wasm modules used by users in middleware https://github.com/opennextjs/opennextjs-netlify/blob/6b56128b90ce9d7d8bc7a6a125fb755f6fe160ac/src/build/functions/edge.ts#L125-L133 

and pass module to `init` function of htmlrewriter (argumentless init doesn't work after vendoring)

Reply:
https://gist.github.com/pieh/b205a685e518ef62afaab392e61e064f this is standalone script using this method that can be tested.

You can also test that passing crap to `init` first arg like:
```
await htmlRewriterInit({
  module_or_path: 'wat',
})
```

would result in errors:

```
error: Uncaught (in promise) TypeError: Invalid URL: 'wat'
    module_or_path = fetch(module_or_path);
                     ^
    at getSerialization (ext:deno_url/00_url.js:98:11)
    at new URL (ext:deno_url/00_url.js:405:27)
    at new Request (ext:deno_fetch/23_request.js:329:25)
    at ext:deno_fetch/26_fetch.js:320:27
    at new Promise (<anonymous>)
    at fetch (ext:deno_fetch/26_fetch.js:316:18)
    at __wbg_init (https://deno.land/x/htmlrewriter@v1.0.0/pkg/htmlrewriter.js:1207:22)
    at file:///Users/misiek/dev/pgs-next-runtime/edge-runtime/test.ts:11:7
```

so the argument has effect. 

Commenting out `init` call also result in errors:

```
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'htmlrewriter_new')
    const ret = wasm.htmlrewriter_new(
                     ^
    at new HTMLRewriter (https://deno.land/x/htmlrewriter@v1.0.0/pkg/htmlrewriter.js:781:22)
    at Object.start (https://deno.land/x/htmlrewriter@v1.0.0/src/index.ts:53:20)
    at Module.invokeCallbackFunction (ext:deno_webidl/00_webidl.js:981:16)
    at new TransformStream (ext:deno_web/06_streams.js:6214:16)
    at HTMLRewriter.transform (https://deno.land/x/htmlrewriter@v1.0.0/src/index.ts:48:29)
    at rewriter (file:///Users/misiek/dev/pgs-next-runtime/edge-runtime/test.ts:39:6)
    at file:///Users/misiek/dev/pgs-next-runtime/edge-runtime/test.ts:43:27
```